Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -79,6 +118,12 @@ class MineField {
     }
     
     
+    /**
+     Returns a given number of random, unique correct coordinates for a Minefield's given size.
+     - parameter count: The amount of random coordinates to get. If higher than actual amount of possible coordinates in the given Minefield's size, will be set to maximum amount of possible coordinates.
+     - parameter size: The size of the Minefield.
+     - returns: An Array containing the requested amount of randomized unique coordinates.
+     **/
     private static func getPlacementOptions(count: Int, forArraySize size: Int) -> [Coordinates] {
```

Comment:
I did a **lot** of reading last night (drugs are bad) 😅 .
The conclusion I reached was that generally:
Single instance > Singleton > Static functions

A singleton is generally better than static functions because we can pass that instance as part of an interface (i.e a classes dependencies can be passed to it within its initialiser rather than instantiated globally else where or within its functions).

While a single instance is better than a singleton because it's easier to test (we can create blank instances of it), people can't access it globally and create hidden dependencies and we can create custom initialisers for it.

Don't take my word for it have a look yourself, if we're going to use or not use a static or singleton pattern the time to decide is now as high as I was last night I think this is a good thing to think about.

Best links I read:
Great [Example of global state](http://misko.hevery.com/2008/08/17/singletons-are-pathological-liars/) sucking balls

[Counter Argument to "Singletons being evil"](http://jorudolph.wordpress.com/2009/11/22/singleton-considerations/) that says when they should be used (doesn't apply for our project)

Discussion on [alternatives to the singleton pattern](https://softwareengineering.stackexchange.com/questions/40373/so-singletons-are-bad-then-what) (I don't think we need dependency injection for this project but if we do then we should learn it)